harmony_netcdf_to_zarr/mosaic_utilities.py

```diff
@@ -42,6 +44,78 @@
                           'd': days_delta}
 
 
+class DimensionInformation:
+    """ A class containing information on a dimension, including the path of
+        the dimension variable within a NetCDF-4 or Zarr Dataset, the values of
+        the 1-dimensional dimension array and any associated temporal epoch
+        and unit. For TRT-121, it is initially assumed that non-temporal
+        dimensions can be aggregated without requiring offsets similar to
+        that stored in the temporal dimensions `units` metdata attribute.
+
+        This class can be used to contain both the dimension information from
+        individual input NetCDF-4 files, as well as the aggregated output
+        dimension.
+
+    """
+    def __init__(self, dimension_path: str, dimension_values: np.ndarray,
+                 dimension_units: str):
+        self.dimension_path = dimension_path
+        self.values = dimension_values
+        self.units = dimension_units
+        self.epoch = None
+        self.time_unit = None
+        self._get_epoch_and_unit()
+
```

Comment:
I don't think we do need special handling for all of these cases. The only reason we need special handling for time is because the values in the array aren't all absolute numbers, they are relative to information in the `units` metadata attribute. We _might_ (but probably won't) need additional special handling for:

* Wrapped vs unwrapped longitudes (but probably not because all granules in a collection should have the same wrapping).
* Possibly vertical dimensions if there is a reference value (again probably not, so far I've only seen physical units, like Pascals or metres, etc, in the `units` metadata attribute for these). 

Generally, I've not seen any modifying information in the `units` metadata to indicate values for non-temporal dimensions within different granules might be incompatible with each other. On the other hand, temporal units, such as for MERRA-2, have individual epochs for each granule. That's why we need the special handling there.

Reply:
Yeah - I am good with this.  It is suggestive that perhaps even the ORNL ABoVE use-case might work.  Extra points for that, if it works (;-)